### PR TITLE
Add cli for staking program

### PR DESCRIPTION
**Pull request by codewithgun**
- Add CLI for staking program
- Box stake function account to avoid stack frame violation

### CODE REVIEW

File: cli-staking/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+fn unstake(
+    program: &Program,
+    base: &String,
+    token_mint: &Pubkey,
+    payer: &Keypair,
+    unmint_amount: u64,
+) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
Comment (andrewsource147) [inline cli-staking/src/main.rs:124]: Ditto^

File: cli-staking/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        .signer(payer);
+
+    let signature = builder.send()?;
+    println!("Signature {:?}", signature);
+
+    Ok(())
+}
+
+fn show_vault_info(program: &Program, base: &String, token_mint: &Pubkey) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
Comment (andrewsource147) [inline cli-staking/src/main.rs:243]: Ditto^

File: cli-staking/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+fn transfer_admin(
+    program: &Program,
+    base: &String,
+    token_mint: &Pubkey,
+    admin: &Keypair,
+    new_admin: &Pubkey,
+) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
Comment (andrewsource147) [inline cli-staking/src/main.rs:267]: Ditto^

File: cli-staking/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+fn update_locked_reward_degradation(
+    program: &Program,
+    base: &String,
+    token_mint: &Pubkey,
+    admin: &Keypair,
+    locked_reward_degradation: u64,
+) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
[the remark this answers is not in the document]
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

File: cli-staking/src/main.rs (hunk 1 above)
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

File: cli-staking/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+fn reward(
+    program: &Program,
+    base: &String,
+    token_mint: &Pubkey,
+    payer: &Keypair,
+    amount: u64,
+) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
[the remark this answers is not in the document]
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

File: cli-staking/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+}
+
+fn stake(
+    program: &Program,
+    base: &String,
+    token_mint: &Pubkey,
+    payer: &Keypair,
+    amount: u64,
+) -> Result<()> {
+    let base_keypair = read_keypair_file(base).expect("Base keypair file not found");
```
[the remark this answers is not in the document]
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

File: cli-staking/src/main.rs (hunk 2 above)
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

File: cli-staking/src/main.rs (hunk 3 above)
Comment (codewithgun) [reply cli-staking/src/main.rs]: Fixed

**Review by andrewsource147 — APPROVED** [review]

**Pull request merged**